### PR TITLE
Encapsulating work with Designer News in a repo.

### PR DESCRIPTION
**Pull request by florina-muntenescu**
Removing DesignerNews from BaseDataManager and using it only where needed.

### CODE REVIEW

File: app/src/main/java/io/plaidapp/data/LoadSourceCallback.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package io.plaidapp.data
```
Comment (nickbutcher) [inline app/src/main/java/io/plaidapp/data/LoadSourceCallback.kt]: Copyright notice.

File: app/src/main/java/io/plaidapp/data/LoadSourceCallback.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+interface LoadSourceCallback {
+
+    fun sourceLoaded(result: List<PlaidItem>?, page: Int, source: String)
+
+    fun loadFailed(source: String)
+}
```
Comment (nickbutcher) [inline app/src/main/java/io/plaidapp/data/LoadSourceCallback.kt]: nit: newline

File: app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
+package io.plaidapp.data.api.designernews
```
Comment (nickbutcher) [inline app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt]: Copyright

File: app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                } else {
+                    callback.loadFailed(SourceManager.SOURCE_DESIGNER_NEWS_POPULAR)
+                }
+            }
+
+            override fun onFailure(call: Call<List<Story>>, t: Throwable) {
+                callback.loadFailed(SourceManager.SOURCE_DESIGNER_NEWS_POPULAR)
+            }
+        })
+        inflight[SourceManager.SOURCE_DESIGNER_NEWS_POPULAR] = topStories
```
Comment (nickbutcher) [inline app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt]: Shouldn't the repo remove from the `inflight` map on response/failure?

File: app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    fun cancelAllRequests() {
+        for (request in inflight.values) request.cancel()
+    }
+
+    fun cancelRequestOfSource(source: String) {
+        inflight[source].apply { this?.cancel() }
+    }
+}
```
Comment (nickbutcher) [inline app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt]: nit: newline

**Comment by todemudia** [comment]
On Jun 6, 2018 4:34 PM, "Nick Butcher" <notifications@github.com> wrote:

*@nickbutcher* commented on this pull request.
------------------------------

In app/src/main/java/io/plaidapp/data/LoadSourceCallback.kt
<https://github.com/nickbutcher/plaid/pull/253#discussion_r193454316>:

> @@ -0,0 +1,8 @@
+package io.plaidapp.data
+
+interface LoadSourceCallback {
+
+    fun sourceLoaded(result: List<PlaidItem>?, page: Int, source: String)
+
+    fun loadFailed(source: String)
+}

nit: newline
------------------------------

In
app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt
<https://github.com/nickbutcher/plaid/pull/253#discussion_r193454779>:

> @@ -0,0 +1,83 @@
+package io.plaidapp.data.api.designernews

Copyright
------------------------------

In
app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt
<https://github.com/nickbutcher/plaid/pull/253#discussion_r193455993>:

> +    fun loadTopStories(page: Int, callback: LoadSourceCallback) {
+        val topStories = service.getTopStories(page)
+        topStories.enqueue(object : Callback<List<Story>> {
+            override fun onResponse(call: Call<List<Story>>,
response: Response<List<Story>>) {
+                if (response.isSuccessful) {
+                    callback.sourceLoaded(response.body(), page,
SourceManager.SOURCE_DESIGNER_NEWS_POPULAR)
+                } else {
+
callback.loadFailed(SourceManager.SOURCE_DESIGNER_NEWS_POPULAR)
+                }
+            }
+
+            override fun onFailure(call: Call<List<Story>>, t: Throwable) {
+                callback.loadFailed(SourceManager.SOURCE_DESIGNER_NEWS_POPULAR)
+            }
+        })
+        inflight[SourceManager.SOURCE_DESIGNER_NEWS_POPULAR] = topStories

Shouldn't the repo remove from the inflight map on response/failure?
------------------------------

In
app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt
<https://github.com/nickbutcher/plaid/pull/253#discussion_r193456129>:

> +            override fun onFailure(call: Call<List<Story>>, t: Throwable) {
+                callback.loadFailed(query)
+            }
+        })
+
+        inflight[query] = searchCall
+    }
+
+    fun cancelAllRequests() {
+        for (request in inflight.values) request.cancel()
+    }
+
+    fun cancelRequestOfSource(source: String) {
+        inflight[source].apply { this?.cancel() }
+    }
+}

nit: newline

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/nickbutcher/plaid/pull/253#pullrequestreview-126429939>,
or mute the thread

File: app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt (hunk 4 above)
Comment (florina-muntenescu) [reply app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt]: for now I just moved the implementation from one class to another. No changes in logic done.

**Comment by florina-muntenescu** [comment]
call removed. @nickbutcher ready to review

**Pull request merged**